src/orion/core/worker/experiment.py

```diff
@@ -615,35 +650,101 @@ def stats(self):
+        # (from a start time to an end time or heartbeat)
+        intervals = []
+        for trial in trials:
+            interval = trial.execution_interval
+            if interval:
+                intervals.append(interval)
+        if intervals:
+            min_start_time = min(interval[0] for interval in intervals)
+            max_end_time = max(interval[1] for interval in intervals)
+            duration = max_end_time - min_start_time
+        else:
+            duration = datetime.timedelta()
+
+        # Compute ETA
+        if not self.max_trials or math.isinf(self.max_trials):
+            # If max_trials is None, 0 or infinite, we cannot compute ETA
+            eta = None
+        elif len(completed_trials) > self.max_trials:
+            # If there are more completed trials than max trials, then ETA should be 0 (?)
+            eta = datetime.timedelta()
+        elif not completed_trials:
+            # If there are no completed trials, then we set ETA to infinite
+            # NB: float("inf") may lead to wrong JSON syntax, so we just write "infinite"
+            eta = "infinite"
+        else:
+            # Compute ETA using duration of completed trials
+            completed_intervals = [
+                trial.execution_interval for trial in completed_trials
+            ]
+            min_start_time = min(interval[0] for interval in completed_intervals)
+            max_end_time = max(interval[1] for interval in completed_intervals)
+            completed_duration = max_end_time - min_start_time
+            eta = (completed_duration / len(completed_trials)) * (
+                self.max_trials - len(completed_trials)
+            )
+
+        # Compute progress
+        if self.max_trials is None or math.isinf(self.max_trials):
+            progress = None
+        else:
+            progress_base = max(self.max_trials, len(trials))
+            if progress_base == 0:
+                progress = None
+            else:
+                progress = len(completed_trials) / progress_base
 
         return ExperimentStats(
-            trials_completed=trials_completed,
+            trials_completed=len(completed_trials),
             best_trials_id=best_trials_id,
             best_evaluation=best_evaluation,
             start_time=start_time,
             finish_time=finish_time,
             duration=duration,
+            whole_clock_time=sum(
+                (trial.duration for trial in trials),
+                datetime.timedelta(),
+            ),
+            nb_trials=len(trials),
+            eta=eta,
+            eta_milliseconds=eta.total_seconds() * 1000
+            if isinstance(eta, datetime.timedelta)
+            else None,
+            trial_status_count={**Counter(trial.status for trial in trials)},
+            progress=progress,
+            max_trials=(
+                "infinite"
+                if self.max_trials is not None and math.isinf(self.max_trials)
```

Comment:
```suggestion
                if math.isinf(self.max_trials)
```
Seems to me if second clause is true and first one is necessarily true no?

Reply:
Function `math.isinf()` expects only a number, not None. So, if `max_trials` is None, then `math.isinf(max_trials)` will raise an exception. Thus, I need to check `max_trials` before calling `math.isinf()`.

```
>>> math.isinf(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: must be real number, not NoneType
```

Reply:
Aaaah, I see. Thanks!